docstrfmt/main.py

```diff
@@ -484,7 +511,22 @@ def leave_FunctionDef(
         self._object_names.pop(-1)
         return updated_node
 
-    def leave_SimpleString(
+    def _escape_quoting(self, node: SimpleString) -> SimpleString:
+        """Escapes quotes in a docstring when necessary."""
+        # handles quoting escaping once
+        for quoting in ('"""', '"', "'''", "'"):
+            if node.value.startswith(quoting) and node.value.endswith(quoting):
+                inner_value = node.value[len(quoting) : -len(quoting)]
+                if quoting in inner_value:
+                    node = node.with_changes(
+                        value=quoting
+                        + inner_value.replace(quoting[0], "\\" + quoting[0])
```

Comment:
Escape is only needed every `quoting` so it should look like this:
```suggestion
                        + inner_value.replace(quoting, "\\" + quoting)
```

Reply:
You sure? Quoting can be either simple or triple, even for docstrings.  So, the code here counted on this use-case as well.

Reply:
OK - the `_is_docstring` method only applies to triple-quoted strings - it is true that the above code can be simplified.

Reply:
Frankly, I don't want to change the guard you had before on your code (https://github.com/LilSpazJoekp/docstrfmt/blob/master/docstrfmt/main.py#L463) as that may have some repercussions, and it is not strictly part of this feature.